react/features/base/react/components/web/MultiSelectAutocomplete.js

```diff
@@ -0,0 +1,316 @@
+         * The function called when the selection changes.
+         */
+        onSelectionChange: React.PropTypes.func,
+
+        /**
+         * The placeholder text of the input component.
+         */
+        placeholder: React.PropTypes.string,
+
+        /**
+         * The service providing the search.
+         */
+        resourceClient: React.PropTypes.shape({
+            makeQuery: React.PropTypes.func,
+            parseResults: React.PropTypes.func
+        }).isRequired,
+
+        /**
+         * Indicates if the component should fit the container.
+         */
+        shouldFitContainer: React.PropTypes.bool,
+
+        /**
+         * Indicates if we should focus.
+         */
+        shouldFocus: React.PropTypes.bool
+    };
+
+    /**
+     * Initializes a new {@code MultiSelectAutocomplete} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        const defaultValue = this.props.defaultValue || [];
+
+        this.state = {
+            /**
+             * Indicates if the dropdown is open.
+             */
+            isOpen: false,
+
+            /**
+             * The filter value.
+             */
+            filterValue: '',
+
+            /**
+             * Indicates if the component is currently loading results.
+             */
+            loading: false,
+
+
+            /**
+             * Indicates if there was an error.
+             */
+            error: false,
+
+            /**
+             * The list of result items.
+             */
+            items: [],
+
+            /**
+             * The list of selected items.
+             */
+            selectedItems: [ ...defaultValue ] || []
+        };
+
+        this._onFilterChange = this._onFilterChange.bind(this);
+        this._onRetry = this._onRetry.bind(this);
+        this._onSelectionChange = this._onSelectionChange.bind(this);
+        this._sendQuery = this._sendQuery.bind(this);
+    }
+
+    /**
+     * Clears the selected items.
+     *
+     * @returns {void}
+     */
+    clear() {
+        this.setState({
+            selectedItems: []
+        });
+    }
+
+    /**
+     * Sets the state and sends a query on filter change.
+     *
+     * @param {string} filterValue - The filter text value.
+     * @private
+     * @returns {void}
+     */
+    _onFilterChange(filterValue) {
+        this.setState({
+            filterValue,
+            isOpen: Boolean(this.state.items.length)
+                && Boolean(filterValue),
+            items: filterValue ? this.state.items : []
+        }, this._sendQuery);
+    }
+
+    /**
+     * Retries the query on retry.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onRetry() {
+        this._sendQuery();
+    }
+
+    /**
+     * Sends a query to the resourceClient.
+     *
+     * @returns {void}
+     */
+    _sendQuery() {
```

Comment:
What I meant is that this is WiP and it won't be merged without this being fixed in one way or another!